Tasks/DotNetCoreInstallerV0/externals/get-os-platform.ps1

```diff
@@ -0,0 +1,18 @@
+function Get-Machine-Architecture()
+{
+    # possible values: AMD64, IA64, x86
+    return $ENV:PROCESSOR_ARCHITECTURE
```

Comment:
This doesn't tell you the architecture of the processor but only of the process with the environment variable.
We need to fetch the Machine architecture here given that Machine and Agent architecture can never mismatch.

Reply:
This will detect the arch of OS. machine and agent arch can mismatch - i can install x86 agent on x64 machine/os. I am following what dotnet install scripts do

Reply:
This make sense but still I have many confusion around it. Its fine if you are following the dotnet install script's logic.